Microsoft.Toolkit.Mvvm/Input/AsyncRelayCommand{T}.cs

```diff
@@ -19,17 +19,17 @@ public sealed class AsyncRelayCommand<T> : ObservableObject, IAsyncRelayCommand<
         /// <summary>
         /// The <see cref="Func{TResult}"/> to invoke when <see cref="Execute(T)"/> is used.
         /// </summary>
-        private readonly Func<T, Task>? execute;
+        private readonly Func<T?, Task>? execute;
 
         /// <summary>
         /// The cancelable <see cref="Func{T1,T2,TResult}"/> to invoke when <see cref="Execute(object?)"/> is used.
         /// </summary>
-        private readonly Func<T, CancellationToken, Task>? cancelableExecute;
+        private readonly Func<T?, CancellationToken, Task>? cancelableExecute;
 
         /// <summary>
         /// The optional action to invoke when <see cref="CanExecute(T)"/> is used.
         /// </summary>
-        private readonly Func<T, bool>? canExecute;
+        private readonly Predicate<T?>? canExecute;
```

Comment:
What's the difference here of Predicate vs. Func?

Reply:
They're essentially identical, the only difference is they are two different delegate types.

They're defined like this in the BCL:

```csharp
public delegate bool Predicate<in T>(T obj);
public delegate TResult Func<in T, out TResult>(T arg);
```

So they're effectively the same in this context, I just picked `Predicate<T>` as I thought the name was more expressive, and because the returned `bool` parameter is part of the delegate declaration so you don't need to manually type that out.

I can be convinced to use either if you have a strong preference for one or the other 🙂

Reply:
Fine either way if it's functionally equivalent, was just curious. 🙂